java/arcs/core/data/CapabilityNew.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+import arcs.core.util.Time
+
+sealed class CapabilityNew {
+    enum class Comparison { LessStrict, Equivalent, Stricter }
+
+    fun isEquivalent(other: CapabilityNew): Boolean {
+        return compare(other) == Comparison.Equivalent
+    }
+    fun contains(other: CapabilityNew) = isEquivalent(other)
+    fun isLessStrict(other: CapabilityNew) = compare(other) == Comparison.LessStrict
+    fun isSameOrLessStrict(other: CapabilityNew) = compare(other) != Comparison.Stricter
+    fun isStricter(other: CapabilityNew) = compare(other) == Comparison.Stricter
+    fun isSameOrStricter(other: CapabilityNew) = compare(other) != Comparison.LessStrict
+
+    abstract fun compare(other: CapabilityNew): Comparison
+
+    data class Persistence(val type: Type) : CapabilityNew() {
```

Comment:
`Kind` instead of `Type`?

Reply:
aren't type and kind synonyms? or do you want to avoid confusion with the Arcs `Type` in general? 
i'll have to rename the property in the TS class as well, so just want to doublecheck with you first.

Reply:
I was mainly trying to avoid confusion with Arcs `Type`. Arcs `Type` is what popped into my mind when I saw it first.

> i'll have to rename the property in the TS class as well, so just want to doublecheck with you first.

Is that code in this PR as well? (Just making sure this is not a lot of work.)

Reply:
that code is already submitted `capabilities.ts`, but it shouldn't be a problem, i'll rename now.

Reply:
done